packages/common-server/src/logger.ts

```diff
@@ -50,6 +51,28 @@ function createLogger(
   }
 }
 
+/** Create a logger. The logger **must** be disposed after being used if the function returned a dispose callback, otherwise it will leak file descriptors and may lead to crashes. */
+function createDisposableLogger(
```

Comment:
to reduce state, can we always return a `dispose()` function? for `stdout` case, it should just be a null-op